source/tests/system/nitclk_driver_api_tests.cpp

```diff
@@ -0,0 +1,268 @@
+  }
+
+  void set_session_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id, const int session_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::SetAttributeViSessionRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    request.mutable_value()->set_id(session_id);
+    tclk::SetAttributeViSessionResponse response;
+
+    ::grpc::Status status = GetTClkStub()->SetAttributeViSession(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+  }
+
+  void set_string_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id, const ViString attribute_value)
+  {
+    ::grpc::ClientContext context;
+    tclk::SetAttributeViStringRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    request.set_value(attribute_value);
+    tclk::SetAttributeViStringResponse response;
+
+    ::grpc::Status status = GetTClkStub()->SetAttributeViString(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+  }
+
+ private:
+  std::shared_ptr<::grpc::Channel> channel_;
+  std::unique_ptr<::nidevice_grpc::Session> scope_session_;
+  std::unique_ptr<scope::NiScope::Stub> niscope_stub_;
+  std::unique_ptr<tclk::NiTClk::Stub> nitclk_stub_;
+  std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+  std::unique_ptr<scope::NiScopeLibrary> niscope_library_;
+  std::unique_ptr<scope::NiScopeService> niscope_service_;
+  std::unique_ptr<tclk::NiTClkLibrary> nitclk_library_;
+  std::unique_ptr<tclk::NiTClkService> nitclk_service_;
+  std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiTClkDriverApiTest, ConfigureForHomogeneousTriggers_ReturnsStatusAsSuccessful)
+{
+  ::grpc::ClientContext context;
+  tclk::ConfigureForHomogeneousTriggersRequest request;
+  request.add_sessions()->set_id(GetScopeSessionId());
+  tclk::ConfigureForHomogeneousTriggersResponse response;
+  ::grpc::Status status = GetTClkStub()->ConfigureForHomogeneousTriggers(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+}
+
+TEST_F(NiTClkDriverApiTest, SetupForSyncPulseSenderSynchronize_ReturnsStatusAsSuccessful)
+{
+  ::grpc::ClientContext context;
+  tclk::SetupForSyncPulseSenderSynchronizeRequest request;
+  request.add_sessions()->set_id(GetScopeSessionId());
+  tclk::SetupForSyncPulseSenderSynchronizeResponse response;
+  ::grpc::Status status = GetTClkStub()->SetupForSyncPulseSenderSynchronize(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+}
+
+TEST_F(NiTClkDriverApiTest, SetAttributeViReal64_GetAttributeViReal64ReturnsSameValue)
```

Comment:
Nitpick: I think these attribute tests should be restructured and named per `Given When Then`. In this case, maybe `SetAttributeViReal64_CallGetAttributeViReal64_ValuesMatch` where setting the attribute is the setup, getting the attribute value is the action, and confirming the values match is the assertion. I'm not sure that applies cleanly here, however, since the set and get are both actions being tested by this.